projects/stef.md

```diff
@@ -0,0 +1,330 @@
+# Otel/STEF
+
+## Background and description
+
+Otel/STEF (Sequential Tabular Encoding Format) is a new data format and network 
+protocol for OpenTelemetry data.
+
+On the tested datasets Otel/STEF payloads are several times more compact than OTLP, with 
+comparable serialization speed. Otel/STEF payloads are also smaller and serialization is 
+faster than Otel Arrow Phase 1.
+
+Otel/STEF targets a narrower niche than OTLP or Otel Arrow and is more efficient
+for that niche. Otel/STEF is optimized for payload size and fast serialization
+and is primarily intended to be used in the last communication leg from Otel Collector
+to observability backend where large volumes of data make network cost
+savings significant.
+
+Otel/STEF supports options that allow trading speed for size. At the highest 
+speed setting it can be faster than OTLP (_including_ conversion to/from pdata), while 
+producing significantly smaller payloads.
+
+A draft specification and a prototype implementation of Otel/STEF is attached to thi
+proposal.
+
+### Goals, objectives, and requirements
```

Comment:
I think there should be two additional goals here (possibly one if you phrase it well).

- We should evaluating an efficient file-based protocol for direct-export from API with out-of-band "Collector-style" SDK.  That is, imagine an SDK implementation that can serialize events out-of-band quickly and efficiently, and offers more resilience on process-death.   This is what I was working my towards with https://github.com/jsuereth/otlp-mmap/ and I think if we invest in a stateless protocol, this should be a use case it can support.
- Providing guidance to eBPF based telemetry extraction.  If we are able to define structures and buffers and efficient stateful communication, we might be able to provide a good set of primitives for eBPF based event-extraction (complementary to my first bullet point).   This is an avenue I think may be worth exploring.

Reply:
>We should evaluating an efficient file-based protocol for direct-export from API with out-of-band "Collector-style" SDK.

That's certainly something we can extend STEF to do. A mmap-ed ring buffer of STEF frames can be that. Since STEF optionally allows full state resets between frames, it essentially has a stateless mode built-in (with resets happening every frame).

>Providing guidance to eBPF based telemetry extraction.

I am not sure I understand this one.

Reply:
>I think there should be two additional goals here (possibly one if you phrase it well).

I added a more generic goal that says that the project should evaluate additional use-cases.